uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -259,10 +260,13 @@ define([
     function checkIfBalanceChangeHasChanged () {
       if (!vm.isMode('edit') || vm.isRole('staff') || getLeaveType() === 'toil') { return; }
 
-      return vm.request.calculateBalanceChange(vm.selectedAbsenceType.calculation_unit_name)
-        .then(function (balanceChange) {
-          if (+vm.balance.change.amount !== +balanceChange.amount) {
-            promptBalanceChangeRecalculation(balanceChange);
+      return vm.request.checkIfBalanceChangeNeedsForceRecalculation()
```

Comment:
I think this would read better, but I don't think @AkA84 would agree:

```js
return vm.request.ifBalanaceChangeNeedsRecalculation()
  .then(LeaveRequestService.promptIfProceedWithBalanceChangeRecalculation)
  .then(function () {
    $rootScope.$emit('LeaveRequestPopup::recalculateBalanceChange');
  });
```

If `dialog.open` rejects a promise when the prompt is cancelled, why can't we reject a promise if a condition is not met?

Reply:
If `balanceChangeHasBeenChanged` is true, it always needs to be rejected. I get your point, but I would leave it as is for now, since `LeaveRequestService.promptIfProceedWithBalanceChangeRecalculation` does not reject now and that would mean new code, new tests etc. This may increase the PR scope again and this PR is already pretty big for its task.